x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/GenerateUniqueIndexNameStep.java

```diff
@@ -74,7 +74,7 @@ public ClusterState performAction(Index index, ClusterState clusterState) {
         Builder newLifecycleState = LifecycleExecutionState.builder(lifecycleState);
         String policyName = indexMetadata.getLifecyclePolicyName();
         String generatedIndexName = generateIndexName(prefix, index.getName());
-        ActionRequestValidationException validationException = validateGeneratedIndexName(generatedIndexName, clusterState);
+        ActionRequestValidationException validationException = validateGeneratedIndexName(generatedIndexName, clusterState.projectState());
```

Comment:
A follow-up PR will take care of passing the correct project state here instead of using this deprecated method.